integration-tests/configs/otel-gen-client/Dockerfile

```diff
@@ -0,0 +1,11 @@
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+FROM golang:1.21 as build
+WORKDIR /app/
+COPY go.mod go.sum ./
+RUN go mod download
+COPY ./integration-tests/configs/otel-gen-client/ ./
+RUN CGO_ENABLED=0 go build -o main main.go
```

Comment:
Nit: do you need `CGO_ENABLED=0` here? I don't see immediately why we'd need that.

Reply:
yeah because if I build on a mac and then try to run it on Alpine there are problems with C library (apparently musl vs glibc). I tried removing this flag and the binary would not run in docker